dot/peerset/peerset.go

```diff
@@ -729,3 +769,29 @@ func (ps *PeerSet) doWork(ctx context.Context) {
 		}
 	}
 }
+
+func (ps *PeerSet) periodicallyAllocateSlots(ctx context.Context) {
+	ticker := time.NewTicker(ps.nextPeriodicAllocSlots)
+
+	defer func() {
+		ticker.Stop()
+		close(ps.resultMsgCh)
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			if err != nil {
+				logger.Debugf("peerset context finished with error: %w", err)
+			}
+			return
+		case <-ticker.C:
+			for i := 0; i < ps.peerState.getSetLength(); i++ {
+				if err := ps.allocSlots(i); err != nil {
+					logger.Warnf("failed to do action on peerSet: %s", err)
```

Comment:
nit this seems more informative/updated I think:

```suggestion
					logger.Warnf("failed to allocate slots: %s", err)
```